Assignment 3.Rmd

```diff
@@ -127,27 +137,96 @@ Now we can plot our line plot using the ggplot command, "ggplot()".
 ```
 
 Look at the number of people in each cluster, now repeat this process for 3 rather than 2 clusters. Which cluster grouping do you think is more informative? Write your answer below:
 
-Once you have done this, save both of your plots to the Assignment 5 file.
+I think it is more informative when cluster equals 2.When clsuter equals 3,there are 2 groups of the graphs which are very similar.
+
+```{r}
+fit3 <- kmeans(x = K3,centers = 3)
+K4 <- data.frame(K3,fit3$cluster)
+names(K4)[6]<- "cluster"
+head(K4)
+K5 <- gather(data = as.tbl(K4),key = "week","motivation",-cluster)
+K6 <- K5 %>% group_by(week,cluster) %>% summarise(avg=mean(motivation))
+K6$week <- as.numeric(as.factor(K6$week))
+K6$cluster <- as.factor(K6$cluster)
+ggplot(data=K6,aes(x=week,y=avg,col=cluster))+geom_line()+xlab("Week")+ylab("Average Motivation")
+K7 <- K4 %>% group_by(cluster) %>% summarise(cnt=n())
+```
+
+
+Once you have done this, save both of your plots to the Assignment 5 file. Create a Zotero item to index your new computer program (Assignment 5.rmd) in Zotero. Then commit your assignment, push it to your Github account and then Pull request your version to the original assignment version so I can see it.
 
 ##Part II
 
 Now, try to do the same for the data collected in class. Create two sets of clusters, the answers to the questions and regions where people grew up. 
 
+```{r message=FALSE, warning=FALSE, paged.print=FALSE}
+# upload data set
+hudk <- read.csv("/Users/yangliping/my-repo/HUDK4050/assignment3/HUDK405018-cluster.csv",header = FALSE,skip = 3)
+# select answers to the questions and regions where people grew up
+df <- hudk[,c(18:26)]
+# drop missing values
+df <- na.omit(df)
+# named data 
+names(df) <- c("city","country","Q1","Q2","Q3","Q4","Q5","Q6","Q7")
+# scale data
+df2 <- scale(df[,-c(1,2)])
+# look at data 
+head(df2)
+# K-Modes Clustering with k=2
+km2 <- kmeans(df2,centers = 2)
+km2
+# k=2
+df$cluster2 <-  as.numeric(km2$cluster)
+# K-Modes Clustering with k=3
+km3 <- kmeans(df2,centers = 3)
+km3
+# k=3
+df$cluster3 <- as.numeric(km3$cluster)
+# transform data
+df3 <-  df %>% 
+  gather(key = "question","answer",Q1:Q7)
+# question to num
+df3$question <- as.numeric(as.factor(df3$question))
+df3$cluster2 <- as.factor(df3$cluster2)
+df3$cluster3 <- as.factor(df3$cluster3)
+# plot k=2
+df3 %>% group_by(question,cluster2)%>%
+  summarise(avg=mean(answer)) %>%
+  ggplot(aes(x=question,y=avg,col=cluster2))+geom_line()+xlab("question")+ylab("mean answer")
+# plot k=3
+df3 %>% group_by(question,cluster3)%>%
+  summarise(avg=mean(answer)) %>%
+  ggplot(aes(x=question,y=avg,col=cluster3))+geom_line()+xlab("question")+ylab("mean answer")
+```
+
+
 ##Part III
 
-Create a visualization that shows the overlap between the two groups of clusters you created in part III.
+Create a visualization that shows the overlap between the two groups of clusters you created in part II.
+
+There are no overlap between two groups of clusters.
```

Comment:
I don't think that is true? But you have created a good plot to demonstrate this.